NinjaResidence/GAMESCENE.cpp

```diff
@@ -27,40 +27,24 @@ GameScene::~GameScene()
 
 SCENE_NUM  GameScene::Update()
 {
+	m_pGameChara->prevSaveMapCharaPos();
 		if (m_pDirectX->GetKeyStatus(DIK_UP))
 		{
-			m_pMapChip->setMapScrollY (m_pGameChara->KeyOperation(UP));
+			m_pGameChara->KeyOperation(UP);
 		}
 		if (m_pDirectX->GetKeyStatus(DIK_DOWN))
 		{
-			m_pMapChip->setMapScrollY (m_pGameChara->KeyOperation(DOWN));
+			m_pGameChara->KeyOperation(DOWN);
 		}
 		if (m_pDirectX->GetKeyStatus(DIK_LEFT))
 		{
-			m_pMapChip->setMapScrollX (m_pGameChara->KeyOperation(LEFT));
+			m_pGameChara->KeyOperation(LEFT);
 		}
 		if (m_pDirectX->GetKeyStatus(DIK_RIGHT))
 		{
-			m_pMapChip->setMapScrollX (m_pGameChara->KeyOperation(RIGHT));
+			m_pGameChara->KeyOperation(RIGHT);
 		}
-		if (m_pDirectX->GetKeyStatus(DIK_W))
-		{
-			m_pMapChip->KeyOperation(UP);
-		}
-		if (m_pDirectX->GetKeyStatus(DIK_S))
-		{
-			m_pMapChip->KeyOperation(DOWN);
-		}
-		if (m_pDirectX->GetKeyStatus(DIK_A))
-		{
-			m_pMapChip->KeyOperation(LEFT);
-		}
-		if (m_pDirectX->GetKeyStatus(DIK_D))
-		{
-			m_pMapChip->KeyOperation(RIGHT);
-		}
-
-	m_pGameChara->Update();
+		m_pGameChara->Update();
 	return Getm_NextScene();
```

Comment:
Getterが返すのがメンバの値ということはわかりきっているので
```suggestion
	return GetNextScene();
```
でいいような気がします

Reply:
なおしました